websites/N/NationStates/presence.ts

```diff
@@ -0,0 +1,288 @@
+			presenceData.smallImageText = "World Assembly";
+		} else if (
+			page.startsWith("/page=wa") ||
+			page.startsWith("/page=un") ||
+			page.startsWith("/page=list_un")
+		) {
+			// World Assembly
+			presenceData.state = "Attending the World Assembly";
+			presenceData.smallImageKey = "https://i.imgur.com/A09MAL2.png";
+			presenceData.smallImageText = "World Assembly";
+		} else if (page.startsWith("/nation=")) {
+			// Nation page
+
+			const nationid: string = page.match(/(?<=\/nation=)\w+/gi)[0];
+			if (nationid === document.body.getAttribute("data-nname")) {
+				// Self nation page
+				presenceData.state = "Viewing Nation";
+				presenceData.smallImageKey = "https://i.imgur.com/BAzbs0d.png";
+				presenceData.smallImageText = "My Nation";
+			} else {
+				// Other nation page
+				presenceData.state = `Viewing Nation: ${await fetchNationName(
+					nationid
+				)}`;
+				presenceData.smallImageKey = "https://i.imgur.com/hE7TN9d.png";
+				presenceData.smallImageText = "World";
+			}
+		} else if (page.startsWith("/region=")) {
+			// Region page
+			presenceData.state = `Viewing Region: ${await fetchRegionName(
+				page.match(/(?<=\/region=)\w+/gi)[0]
+			)}`;
+			presenceData.smallImageKey = "https://i.imgur.com/hE7TN9d.png";
+			presenceData.smallImageText = "World";
+		} else if (page.startsWith("/page=dispatches")) {
+			// Dispatches
+			presenceData.state = "Browsing Dispatches";
+			presenceData.smallImageKey = "https://i.imgur.com/pArUzoy.png";
+			presenceData.smallImageText = "Dispatches";
+		} else if (page.startsWith("/page=news")) {
+			// News
+			presenceData.state = "Reading News";
+			presenceData.smallImageKey = "https://i.imgur.com/pArUzoy.png";
+			presenceData.smallImageText = "News";
+		} else if (page.startsWith("/page=dispatch")) {
+			// Reading a dispatch
+			presenceData.smallImageKey = "https://i.imgur.com/pArUzoy.png";
+			presenceData.smallImageText = "Dispatches";
+			let title = document
+				.querySelector("#content")
+				?.querySelector(".dispatch")
+				?.querySelector("h2")?.textContent;
+			if (title && title.length > 1) {
+				if (title.length > 128 - 18) title = `${title.substring(0, 128 - 19)}…`;
+				presenceData.state = `Reading Dispatch: ${title}`;
+			} else presenceData.state = "Reading a Dispatch";
+		} else if (page.startsWith("/page=create_dispatch")) {
+			// Writing a dispatch
+			presenceData.state = "Writing Dispatch";
+			presenceData.smallImageKey = "https://i.imgur.com/pArUzoy.png";
+			presenceData.smallImageText = "Dispatches";
+		} else if (
+			page.startsWith("/page=store") ||
+			page.startsWith("/page=cart") ||
+			page.startsWith("/page=orders")
+		) {
+			// Store
+			presenceData.state = "Browsing the Store";
+			presenceData.smallImageKey = "https://i.imgur.com/28eyl2A.png";
+			presenceData.smallImageText = "Store";
+		} else {
+			// Other
+			presenceData.state = "Browsing";
+			delete presenceData.smallImageKey;
+		}
```

Comment:
Maybe improve readablity my using a switch statement?

Reply:
I can rework the chain using 'switch' statement, but it would be a bit inconvenient, as the website NationStates has many pages starting with, for example, "/page=un" - so using the `.startsWith()` method is more convenient than listing every possible page.
But, sure, I can start working on it.

Reply:
The easiest would be to split the pathname. If you need help feel free to react or dm me on discord.

Reply:
Thanks, I'm pushing the new version with switches just now. I think I figured it out, and I hope it's more readable this way.